examples/client/lwm2mclient.c

```diff
@@ -59,12 +59,12 @@
 #include "lwm2mclient.h"
 #include "liblwm2m.h"
 #include "commandline.h"
-#ifdef WITH_TINYDTLS
+#if defined(WITH_TINYDTLS) || defined(WITH_MBEDTLS)
```

Comment:
I think it would make sense to have a generic define "DTLS" being set (and tested for) whenever either of the implementations is in use (checked for).

Also, in CMake, instead of renaming DTLS to DTLS_TINYDTLS, I'd rather change the semantic of DTLS and adding a additional DTLS_TINYDTLS and DTLS_EMBEDTLS options.